prac_09/taxi_simulator.py

```diff
@@ -0,0 +1,75 @@
+"""
+Write a taxi simulator program that uses the Taxi and SilverServiceTaxi classes.
+"""
+from prac_09.taxi import Taxi
+from prac_09.silver_service_taxi import SilverServiceTaxi
+
+MENU = """q)uit, c)hoose taxi, d)rive"""
+
+
+def main():
+    """Each time, until they quit: user can choose a taxi and drive a desired distance.
+    At the end of each trip, the trip cost is displayed and add it to their total bill."""
+    taxis = [Taxi("Prius", 100), SilverServiceTaxi("Limo", 100, 2), SilverServiceTaxi("Hummer", 200, 4)]
+    current_taxi = None
+    total_bill = 0
+
+    print("Let's drive!")
+    print(MENU)
+    choice = input(">>> ").upper()
+    while choice != "Q":
+        if choice == "C":
+            display_taxis(taxis)
+            current_taxi = choose_taxi(current_taxi, taxis)
+        elif choice == "D":
+            total_bill += drive_taxi(current_taxi)
+        else:
+            print("Invalid option")
+        print(MENU)
+        print(f"Bill to date: ${total_bill:.2f}")
+        choice = input(">>> ").upper()
+    print("Thank you.")
+    print(f"Total trip cost: ${total_bill:.2f}")
+    print("Taxis are now:")
+    display_taxis(taxis)
+
+
+def drive_taxi(current_taxi):
+    """user can choose how far they want to drive and handles not choosing a taxi first."""
+    if current_taxi:
```

Comment:
sheesh this is cold, making use of None being the same as false in an if statement. Pretty badass.

Reply:
My ego can't handle all these compliments.